azurelinuxagent/agent.py

```diff
@@ -140,6 +150,21 @@ def run_exthandlers(self, debug=False):
         """
         logger.set_prefix("ExtHandler")
         threading.current_thread().setName("ExtHandler")
+
+        #
+        # Agents < 2.2.53 used to echo the log to the console. Since the extension handler could have been started by
+        # one of those daemons, output a message indicating that output to the console will stop, otherwise users
+        # may think that the agent died if they noticed that output to the console stops abruptly.
+        #
+        # Feel free to remove this code if telemetry shows there are no more agents <= 2.2.53 in the field.
+        #
+        if conf.get_logs_console() and get_daemon_version() < FlexibleVersion("2.2.53"):
+            self.__add_console_appender(logger.LogLevel.INFO)
```

Comment:
Should this be added under the try just to be on the safer side?

Reply:
if the appender was not added we do not want do remove it (nothing bad would happen in the current implementation, but still)

Reply:
I think there's been some confusion, I meant moving the adding part under the try too. Just having a `logger.info` under try adds little value.

```python
 try:
                self.__add_console_appender(logger.LogLevel.INFO)
                logger.info(u"The agent will now check for updates and then will process extensions. Output to /dev/console will be suspended during those operations.")
            finally:
                logger.disable_console_output()
```